### DISCUSSION TITLE
Reduce Testing time with Mysql DB

### DISCUSSION THREAD

**Discussion by AashishAx** [laravel/framework#34987] [Q&A] [answered]
I recently switched my testing environment from in memory sqlite to a mysql DB. As I understand the "RefreshDatabase" checks if we are using in memory DB or an actual DB and skips the migration process if necessary to reduce time. However, right now if I run a test it takes on average over 10 seconds to complete which isn't very ideal. Am I missing something here or is there a way to reduce the time for testing? I have 86 tables in my DB.

I'm using laravel 7.28.4

**Comment by kohlerdominik** [comment, marked as answer]
You mean a singe test? Maybe you may prefer to `use DatabaseTransactions` rather than RefreshDatabase.

[DatabaseTransactions ](https://github.com/laravel/framework/blob/43bea00fd27c76c01fd009e46725a54885f4d2a5/src/Illuminate/Foundation/Testing/DatabaseTransactions.php) uses Transactions to rollback after a test, just as RefreshDatabase does, but it does not drop and create the database in advance of running the test-suite.

I personally do prefer DatabaseTransactions before RefreshDatabase, as it's faster. However, you have to keep track of your database by your own, which can lead to issues.

**Reply by AashishAx** [reply]
That was my understanding as well and I did test with DatabaseTransactions but the time it took was more or less the same (about 10 sec for a single test)

**Reply by AashishAx** [reply]
Not sure what was wrong with it the first time I tried it. But I tried DatabaseTransactions again and its working as expected now. Thanks for the help
